src/Controls/FloatingMenu/FloactionActionMenu.xaml.cs

```diff
@@ -0,0 +1,211 @@
+            }
+        }
+        #endregion
+
+        #region Image
+        public static readonly BindableProperty ImageProperty = BindableProperty.Create(
+            propertyName: nameof(Image),
+            returnType: typeof(ImageSource),
+            declaringType: typeof(FloatingActionMenu),
+            defaultValue: default(ImageSource));
+
+        public ImageSource Image
+        {
+            get
+            {
+                return (ImageSource)GetValue(ImageProperty);
+            }
+            set
+            {
+                SetValue(ImageProperty, value);
+            }
+        }
+        #endregion
+
+        #region Color
+        public static readonly BindableProperty ColorProperty = BindableProperty.Create(
+            propertyName: nameof(Color),
+            returnType: typeof(Color),
+            declaringType: typeof(FloatingActionMenu),
+            defaultValue: Color.Accent);
+
+        public Color Color
+        {
+            get
+            {
+                return (Color)GetValue(ColorProperty);
+            }
+            set
+            {
+                SetValue(ColorProperty, value);
+            }
+        }
+        #endregion
+
+        #region IsOpen
+        public static readonly BindableProperty OpenedProperty = BindableProperty.Create(
+            propertyName: nameof(IsOpen),
+            returnType: typeof(bool),
+            declaringType: typeof(FloatingActionMenu),
+            defaultValue: true,
+            propertyChanged: IsOpenPropertyChanged);
+
+        private static void IsOpenPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            if (bindable != null && newValue != null)
+            {
+                var control = (FloatingActionMenu)bindable;
+                var isOpen = (bool)newValue;
+                control.IsOpenChangedAsync(isOpen);
+            }
+        }
+
+        private async Task IsOpenChangedAsync(bool isOpen)
+        {
+            // RightIconImage.RotateTo(isOpen ? 180 : 0, length: 150);
+            if (Items != null && _originalContentHeight != default)
+            {
+                _isAnimating = true;
+                var animate = new Animation(
+                    callback: d => ItemsLayout.HeightRequest = d,
+                    start: isOpen ? 0 : _originalContentHeight,
+                    end: isOpen ? _originalContentHeight : 0);
+                animate.Commit(
+                    owner: ItemsLayout,
+                    name: "ExpanderAnimation",
+                    length: 150u);
+
+                var animate2 = new Animation(
+                    callback: d => ItemsLayout.Opacity = d,
+                    start: isOpen ? 0 : 1,
+                    end: isOpen ? 1 : 0);
+                animate2.Commit(
+                    owner: ItemsLayout,
+                    name: "ExpanderFadeAnimation",
+                    length: 150u);
+
+                await Task.Delay(200);
```

Comment:
Task.Delay en dur ? ;)
=> regarde ici pour le faire plus joliment :)
https://blog.xamarin.com/building-custom-animations-in-xamarin-forms/